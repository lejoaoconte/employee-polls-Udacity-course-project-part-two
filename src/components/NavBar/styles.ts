import styled from "styled-components";

export const ContainerNavBar = styled.nav`
  width: 100%;
  height: 60px;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  margin-bottom: 30px;
`;

export const MenuItems = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  gap: 20px;
`;

interface NavBarItemsProps {
  openMenu: boolean;
}

export const NavBarItems = styled.ul<NavBarItemsProps>`
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;

  display: flex;
  gap: 15px;

  @media screen and (max-width: 600px) {
    display: ${({ openMenu }) => (openMenu ? "none" : "flex")};

    position: absolute;

    left: 0;
    bottom: 0;

    flex-direction: column;
    align-items: center;
    background: ${({ theme }) => theme.primary};

    width: 100%;
    padding: 20px 0;
    gap: 20px;

    z-index: 1000;

    box-shadow: 0 -3px 18px -3px rgba(31, 31, 31, 0.75);

    animation-name: example;
    animation-duration: 4s;

    @keyframes example {
      from {
        bottom: -200px;
      }
      to {
        bottom: 0;
      }
    }
  }
`;

export const ListItemNav = styled.li`
  float: left;

  a,
  button {
    text-decoration: none;
    font-weight: 800;
    color: ${({ theme }) => theme["primary-text"]};
    font-size: 1.1rem;

    cursor: pointer;

    background: none;
    border: none;

    @media screen and (max-width: 600px) {
      color: ${({ theme }) => theme.background};
    }
  }

  &:hover {
    filter: brightness(0.6);
  }
`;

export const ButtonOpenMenu = styled.button`
  display: none;
  text-decoration: none;
  font-weight: 800;
  color: ${({ theme }) => theme["primary-text"]};
  font-size: 1.5rem;

  cursor: pointer;

  background: none;
  border: none;

  @media screen and (max-width: 600px) {
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;
