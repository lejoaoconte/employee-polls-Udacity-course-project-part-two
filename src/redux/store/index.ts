import { Dispatch } from "redux";

import { configureStore } from "@reduxjs/toolkit";

import reducers from "redux/reducers";
import middleware from "redux/middleware";

const store = configureStore({ reducer: reducers, middleware: middleware });

export interface DispatchProps {
  dispatch: Dispatch;
}

export type RootState = ReturnType<typeof store.getState>;

export default store;
