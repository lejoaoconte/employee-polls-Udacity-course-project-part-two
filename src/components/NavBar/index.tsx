import { connect } from "react-redux";

import { Link, useNavigate } from "react-router-dom";

import { logoutUser } from "redux/actions/authedUser";

import { ContainerNavBar, ListItemNav, NavBarItems } from "./styles";

export function NavBar({ dispatch }: any) {
  const navigate = useNavigate();

  function handleLogout() {
    localStorage.removeItem("authedUser");
    dispatch(logoutUser());
    navigate("/");
  }

  return (
    <ContainerNavBar>
      <img width={50} src="/logo.svg" alt="" />
      <NavBarItems>
        <ListItemNav>
          <Link to="/dashboard">Home</Link>
        </ListItemNav>
        <ListItemNav>
          <Link to="/leaderboard">Leaderboard</Link>
        </ListItemNav>
        <ListItemNav>
          <Link to="/newpoll">New Poll</Link>
        </ListItemNav>
        <ListItemNav>
          <button onClick={handleLogout}>Logout</button>
        </ListItemNav>
      </NavBarItems>
    </ContainerNavBar>
  );
}

export default connect()(NavBar);
