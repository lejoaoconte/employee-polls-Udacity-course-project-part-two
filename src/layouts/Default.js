import { createContext, useState } from "react";

import styled, { ThemeProvider } from "styled-components";

import { darker, white } from "styles/theme";

export const Container = styled.div`
  width: calc(100% - 50px);
  max-width: 1024px;
  margin: 0 auto;
  padding: 0;
`;

export const ContextDefaultTheme = createContext({});

export function Default({ children }) {
  const [theme, setTheme] = useState(true);

  return (
    <ContextDefaultTheme.Provider value={{ setTheme, theme }}>
      <ThemeProvider theme={theme ? darker : white}>
        <Container>{children}</Container>
      </ThemeProvider>
    </ContextDefaultTheme.Provider>
  );
}
