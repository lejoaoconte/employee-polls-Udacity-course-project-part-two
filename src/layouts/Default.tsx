import { createContext, ReactNode, useState } from "react";

import styled, { ThemeProvider } from "styled-components";

import { darker, white as whitetheme } from "styles/theme";

export const Container = styled.div`
  width: calc(100% - 50px);
  max-width: 1024px;
  margin: 0 auto;
  padding: 60px 0;
`;

interface DefaultLayoutProps {
  children: ReactNode;
}

interface ContextDefaultThemeProps {
  theme: boolean;
  setTheme: (value: boolean) => void;
}

export const ContextDefaultTheme = createContext(
  {} as ContextDefaultThemeProps
);

export function Default({ children }: DefaultLayoutProps) {
  const [theme, setTheme] = useState(false);

  return (
    <ContextDefaultTheme.Provider value={{ setTheme, theme }}>
      <ThemeProvider theme={theme ? darker : whitetheme}>
        <Container>{children}</Container>
      </ThemeProvider>
    </ContextDefaultTheme.Provider>
  );
}
