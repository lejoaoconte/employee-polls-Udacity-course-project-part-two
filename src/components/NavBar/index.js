import React, { useContext, useState } from "react";

import { connect } from "react-redux";

import { Link, useNavigate } from "react-router-dom";

import { Switch } from "components/Switch";

import { ContextDefaultTheme } from "layouts/Default";

import { logoutUser } from "redux/actions/authedUser";

import { FaBars } from "react-icons/fa";

import {
  ButtonOpenMenu,
  ContainerNavBar,
  ListItemNav,
  MenuItems,
  NavBarItems,
} from "./styles";

function NavBar({ dispatch, authedUser }: any) {
  const navigate = useNavigate();
  const { setTheme, theme } = useContext(ContextDefaultTheme);
  const [openMenu, setOpenMenu] = useState(false);

  function handleLogout() {
    dispatch(logoutUser());
    navigate("/");
  }

  function handleChangeTheme() {
    setTheme(!theme);
  }

  function handleOpenMenu() {
    setOpenMenu(!openMenu);
  }

  return (
    <ContainerNavBar data-testid="navbar-test">
      <img width={50} src="/logo.svg" alt="" />
      <MenuItems>
        <p>Hi, {authedUser?.name}</p>
        <Switch
          data-testid="navbar-switch-test"
          label="switch"
          onChange={handleChangeTheme}
          checked={theme}
        />
        <NavBarItems openMenu={openMenu}>
          <ListItemNav>
            <Link to="/dashboard">Home</Link>
          </ListItemNav>
          <ListItemNav>
            <Link to="/leaderboard">Leaderboard</Link>
          </ListItemNav>
          <ListItemNav>
            <Link to="/newpoll">New Poll</Link>
          </ListItemNav>
          <ListItemNav>
            <button
              data-testid="navbar-button-logout-test"
              onClick={handleLogout}
            >
              Logout
            </button>
          </ListItemNav>
        </NavBarItems>
        <ButtonOpenMenu
          data-testid="navbar-button-open-test"
          onClick={handleOpenMenu}
        >
          <FaBars />
        </ButtonOpenMenu>
      </MenuItems>
    </ContainerNavBar>
  );
}

function mapStateToProps({ authedUser }: any) {
  return {
    authedUser,
  };
}

export default connect(mapStateToProps)(NavBar);
