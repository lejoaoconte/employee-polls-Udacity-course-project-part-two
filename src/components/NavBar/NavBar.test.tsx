import "@testing-library/jest-dom";

import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import store from "redux/store";
import { Default as Layout } from "layouts/Default";

import NavBar from ".";

describe("NavBar", () => {
  const setup = () => {
    const component = render(
      <Layout>
        <Provider store={store}>
          <MemoryRouter>
            <NavBar />
          </MemoryRouter>
        </Provider>
      </Layout>
    );
    
    expect(component).toMatchSnapshot();

    const navbar = component.getByTestId("navbar-test");
    const logout = component.getByTestId("navbar-button-logout-test");
    const switchComponent = component.getByTestId("navbar-switch-test");
    const openMenu = component.getByTestId("navbar-button-open-test");

    return {
      navbar,
      logout,
      switchComponent,
      openMenu,
      ...component,
    };
  };

  it("Testing render on screen", () => {
    const { navbar }: any = setup();
    expect(navbar).toBeInTheDocument;
  });

  it("Test onClick buttons", () => {
    const { logout, switchComponent, openMenu }: any = setup();

    fireEvent.click(logout);
    fireEvent.click(switchComponent);
    fireEvent.click(openMenu);
  });
});
