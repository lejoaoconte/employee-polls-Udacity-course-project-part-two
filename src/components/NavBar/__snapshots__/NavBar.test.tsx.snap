// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Test onClick buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe MBEGz"
      >
        <nav
          class="sc-eDvSVe imlKEo"
          data-testid="navbar-test"
        >
          <img
            alt=""
            src="/logo.svg"
            width="50"
          />
          <div
            class="sc-jSUZER jPAoVo"
          >
            <div
              class="sc-gswNZR"
            >
              <input
                checked=""
                class="sc-hLBbgP dYsohK"
                data-testid="navbar-switch-test"
                hidden=""
                id="switch"
                type="checkbox"
              />
              <label
                class="sc-dkrFOg eZboOX"
                for="switch"
              />
            </div>
            <ul
              class="sc-gKPRtg eeATmd"
            >
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/dashboard"
                >
                  Home
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/newpoll"
                >
                  New Poll
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <button
                  data-testid="navbar-button-logout-test"
                >
                  Logout
                </button>
              </li>
            </ul>
            <button
              class="sc-ftTHYK jRcvZm"
              data-testid="navbar-button-open-test"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe MBEGz"
    >
      <nav
        class="sc-eDvSVe imlKEo"
        data-testid="navbar-test"
      >
        <img
          alt=""
          src="/logo.svg"
          width="50"
        />
        <div
          class="sc-jSUZER jPAoVo"
        >
          <div
            class="sc-gswNZR"
          >
            <input
              checked=""
              class="sc-hLBbgP dYsohK"
              data-testid="navbar-switch-test"
              hidden=""
              id="switch"
              type="checkbox"
            />
            <label
              class="sc-dkrFOg eZboOX"
              for="switch"
            />
          </div>
          <ul
            class="sc-gKPRtg eeATmd"
          >
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/dashboard"
              >
                Home
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/leaderboard"
              >
                Leaderboard
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/newpoll"
              >
                New Poll
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <button
                data-testid="navbar-button-logout-test"
              >
                Logout
              </button>
            </li>
          </ul>
          <button
            class="sc-ftTHYK jRcvZm"
            data-testid="navbar-button-open-test"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
          </button>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NavBar Testing render on screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe MBEGz"
      >
        <nav
          class="sc-eDvSVe imlKEo"
          data-testid="navbar-test"
        >
          <img
            alt=""
            src="/logo.svg"
            width="50"
          />
          <div
            class="sc-jSUZER jPAoVo"
          >
            <div
              class="sc-gswNZR"
            >
              <input
                checked=""
                class="sc-hLBbgP dYsohK"
                data-testid="navbar-switch-test"
                hidden=""
                id="switch"
                type="checkbox"
              />
              <label
                class="sc-dkrFOg eZboOX"
                for="switch"
              />
            </div>
            <ul
              class="sc-gKPRtg eeATmd"
            >
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/dashboard"
                >
                  Home
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <a
                  href="/newpoll"
                >
                  New Poll
                </a>
              </li>
              <li
                class="sc-iBYQkv dXEyZH"
              >
                <button
                  data-testid="navbar-button-logout-test"
                >
                  Logout
                </button>
              </li>
            </ul>
            <button
              class="sc-ftTHYK jRcvZm"
              data-testid="navbar-button-open-test"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe MBEGz"
    >
      <nav
        class="sc-eDvSVe imlKEo"
        data-testid="navbar-test"
      >
        <img
          alt=""
          src="/logo.svg"
          width="50"
        />
        <div
          class="sc-jSUZER jPAoVo"
        >
          <div
            class="sc-gswNZR"
          >
            <input
              checked=""
              class="sc-hLBbgP dYsohK"
              data-testid="navbar-switch-test"
              hidden=""
              id="switch"
              type="checkbox"
            />
            <label
              class="sc-dkrFOg eZboOX"
              for="switch"
            />
          </div>
          <ul
            class="sc-gKPRtg eeATmd"
          >
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/dashboard"
              >
                Home
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/leaderboard"
              >
                Leaderboard
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <a
                href="/newpoll"
              >
                New Poll
              </a>
            </li>
            <li
              class="sc-iBYQkv dXEyZH"
            >
              <button
                data-testid="navbar-button-logout-test"
              >
                Logout
              </button>
            </li>
          </ul>
          <button
            class="sc-ftTHYK jRcvZm"
            data-testid="navbar-button-open-test"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
          </button>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
